src/pages/home/ReportScreen.js

```diff
@@ -355,11 +356,17 @@ function ReportScreen({
         // is not stored locally yet. If report.reportID exists, then the report has been stored locally and nothing more needs to be done.
         // If it doesn't exist, then we fetch the report from the API.
         if (report.reportID && report.reportID === getReportID(route) && !isLoadingInitialReportActions) {
+            const parentReport = ReportUtils.getParentReport(report);
+
+            if (ReportUtils.isThread(report) && isEmptyObject(parentReport)) {
+                Report.openReport(report.parentReportID);
+            }
+
```

Comment:
Can you explain the logic behind this? i.e. why it's in that if block

Reply:
@s77rt I thought it worth to call new code when the report exists